parsl/executors/high_throughput/interchange.py

```diff
@@ -217,79 +217,68 @@ def _send_monitoring_info(self, monitoring_radio: Optional[MonitoringRadioSender
-
-                elif command_req == "WORKERS":
-                    num_workers = 0
-                    for manager in self._ready_managers.values():
-                        num_workers += manager['worker_count']
-                    reply = num_workers
-
-                elif command_req == "MANAGERS":
-                    reply = []
-                    for manager_id in self._ready_managers:
-                        m = self._ready_managers[manager_id]
-                        idle_since = m['idle_since']
-                        if idle_since is not None:
-                            idle_duration = time.time() - idle_since
-                        else:
-                            idle_duration = 0.0
-                        resp = {'manager': manager_id.decode('utf-8'),
-                                'block_id': m['block_id'],
-                                'worker_count': m['worker_count'],
-                                'tasks': len(m['tasks']),
-                                'idle_duration': idle_duration,
-                                'active': m['active'],
-                                'parsl_version': m['parsl_version'],
-                                'python_version': m['python_version'],
-                                'draining': m['draining']}
-                        reply.append(resp)
-
-                elif command_req.startswith("HOLD_WORKER"):
-                    cmd, s_manager = command_req.split(';')
-                    manager_id = s_manager.encode('utf-8')
-                    logger.info("Received HOLD_WORKER for {!r}".format(manager_id))
-                    if manager_id in self._ready_managers:
-                        m = self._ready_managers[manager_id]
-                        m['active'] = False
-                        self._send_monitoring_info(monitoring_radio, m)
+        if self.command_channel in self.socks and self.socks[self.command_channel] == zmq.POLLIN:
+
+            command_req = self.command_channel.recv_pyobj()
+            logger.debug("Received command request: {}".format(command_req))
+            if command_req == "CONNECTED_BLOCKS":
+                reply = self.connected_block_history
+
+            elif command_req == "WORKERS":
+                num_workers = 0
+                for manager in self._ready_managers.values():
+                    num_workers += manager['worker_count']
+                reply = num_workers
+
+            elif command_req == "MANAGERS":
+                reply = []
+                for manager_id in self._ready_managers:
+                    m = self._ready_managers[manager_id]
+                    idle_since = m['idle_since']
+                    if idle_since is not None:
+                        idle_duration = time.time() - idle_since
                     else:
-                        logger.warning("Worker to hold was not in ready managers list")
-
-                    reply = None
+                        idle_duration = 0.0
```

Comment:
From practice, I assume that `m['idle_since']` is guaranteed to exist here (given that this hasn't changed), but mildly odd to check for `None`ness then.  (Not a comment on you, just, "hmm.")

Moving on, a clarity suggestion/bikeshed to inline the if.  "The point is that `idle_duration` is set":
```python
idle_since = time.time() - idle_since if idle_since else 0.0
```
Truthy comparison is fine here?

Reply:
`None` is a meaningful value here, meaning not idle.

Reply:
Yes; but I was speaking pragmatically.  As in `idle_since` is only otherwise "falsey" if it's 0.0, but presumably idle_since is an absolute point in time, and thus will never be less than `time.time()` (+NTP drift) when the script started.

No matter, just a minor suggestion.